File: java/money/ch08/src/main/java/com/thesoftwaregorilla/tdd/money/Money.java
package com.thesoftwaregorilla.tdd.money;

public abstract class Money {
    protected int amount;

    public static Money dollar(int amount) {
        return new Dollar(amount);
    }

    public static Money franc(int amount) {
        return new Franc(amount);
    }

    public abstract Money times(int multiplier);

    @Override
    public boolean equals(Object object) {
        Money money = (Money) object;
        return getClass().equals(object.getClass()) && amount == money.amount;
    }

    // See my note in the DollarTest class. I added a getter for amount because I had a constructor test.
    public int getAmount() {
        return amount;
    }

}

File: css/ch08/Money/Money.cs
namespace TheSoftwareGorilla.TDD.Money;

public abstract class Money
{
    public int Amount { get; protected set; }

    public static Money Dollar(int amount)
    {
        return new Dollar(amount);
    }

    public static Money Franc(int amount)
    {
        return new Franc(amount);
    }

    public abstract Money Times(int multiplier);

    public override bool Equals(object? obj)
    {
        return obj is Money money && GetType().Equals(obj.GetType()) && Amount == money.Amount;
    }


}